http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/DiskServiceInterp.scala

```diff
@@ -318,6 +318,87 @@ class DiskServiceInterp[F[_]: Parallel](config: PersistentDiskConfig,
 
     } yield ()
 
+  def deleteAllOrphanedDisks(userInfo: UserInfo,
+                             cloudContext: CloudContext.Gcp,
+                             runtimeDiskIds: Vector[DiskId],
+                             appDisksNames: Vector[Option[DiskName]]
+  )(implicit
+    as: Ask[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- as.ask
+      // Get a list of disks that are not attached to any runtime or app
+      disks <- listDisks(
```

Comment:
I noticed we're passing the `userInfo` to the `list(Runtime/Disk/App)` calls. Does this mean that `deleteAllX` will only delete the records that the calling user has access to? If the user is a workspace owner then that should be everything in the workspace; but otherwise it might _not_ be everything in the workspace.

I just checked `deleteAllAppsV2` and it works the same way. I guess only workspace owners have permission to delete the workspace, and the `deleteAll` APIs are only called at workspace deletion time. So I think this is fine -- just calling out the behavior.

Reply:
Yep, this is by design. The goal is to impersonate the workspace owner when making these calls